packages/boot/test/bootstrapTests/addAssets.test.ts

```diff
@@ -21,7 +21,7 @@ const test = anyTest as TestFn<
 const auctioneerPath = 'published.auction';
 
 test.before(async t => {
-  const context = await makeLiquidationTestContext(t);
+  const context = await makeLiquidationTestContext(t, 'bundles/addAssets');
```

Comment:
The downside of separate caches is fewer cache hits. Having one cache per `test.before` is the same as no cache.

Did you observe corruption? I think we can and should fix that in the caching bundler. Would you file an issue with examples of the problem?

Reply:
I observed corruption, where one test reading a partially-built (or build-in-progress) cache cannot parse the truncated JSON files.

I eliminated the corruption by using lockfiles in endojs/endo#1544, but this had the problem that stale lockfiles would deadlock bundle-source (endojs/endo#2163).

endojs/endo#1714 was touted as a way to solve this deadlock, but it reintroduced the original corruption problem.

If we're going to fix this thoroughly while preventing bundle-source running on the same bundle cache in separate processes from stomping on one another, I think we need to go back to my lockfile implementation, and add an expiry time for the generated lockfiles.

@kriskowal ?

Reply:
To try to provoke the corruption I saw, I'm using a single bundleDir for all the tests.  Then I'll file an issue if I can see it failing.

Reply:
A repro would be good. I was unaware that the lockfile was addressing a corruption issue that using an atomic posix rename upon write-completion wouldn’t solve. I’m hoping that this is just a matter of applying the atomic rename strategy harder.

Reply:
I believe that atomic rename still has a race because there are two files in question: the `-js-meta.json` file and the `.js` file.  I haven't convinced myself that all the possible interleavings of two separate processes updating the same bundle will work out correctly.

Reply:
I’m not sure myself. We should have spaghetti with chopsticks some time. I think the trick is usually to be sure you read, write, and stat the two files in the same order when reading or writing. There might be a retry loop in some cases. Is the concurrency due to Ava? Are we writing and reading bundles from disk concurrently in tests?

Reply:
Yes, in this case, the concurrency is due to AVA.  I'm loathe to reduce AVA's concurrency since that's a bottleneck for agoric-sdk CI testing performance.